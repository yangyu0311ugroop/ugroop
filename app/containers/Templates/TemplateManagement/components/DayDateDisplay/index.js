/**
 * Created by stephenkarpinskyj on 25/5/18.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import moment from 'moment';
import classNames from 'classnames';
import resaga from 'resaga';
import { THE_DOT } from 'appConstants';
import { dateDisplay } from 'utils/constant';
import { NODE_HELPERS } from 'utils/helpers/nodes';
import { H4 } from 'viewComponents/Typography';
import { withStyles } from 'components/material-ui';
import GridContainer from 'components/GridContainer';
import GridItem from 'components/GridItem';
import { CONFIG_IDS, CONFIG } from './config';

const FONT_SIZE = 23;
const style = ({ breakpoints }) => ({
  day: {
    margin: 'unset',
    fontWeight: 600,
    [breakpoints.up('md')]: {
      fontSize: FONT_SIZE,
    },
  },
  dot: {
    margin: '0 8px',
    fontWeight: 800,
    fontSize: 15,
    color: '#86A6EB',
    maxHeight: 20,
  },
  date: {
    margin: 'unset',
    [breakpoints.up('md')]: {
      fontSize: FONT_SIZE,
    },
  },
  grow: {
    flex: 1,
  },
});

export class DayDateDisplay extends PureComponent {
  calculateDate = () => {
    const { startTime } = this.props;
    return moment.utc(startTime);
  };

  renderDayOnly = (classes, day) => {
    const { customClass } = this.props;
    const { day: dayClass } = customClass;
    return (
      <GridContainer spacing={0}>
        <GridItem>
          <H4 className={classNames(classes.day, dayClass)}>{day}</H4>
        </GridItem>
      </GridContainer>
    );
  };

  renderDayAndDate = (classes, day, date) => {
    const { displayInfo } = this.props;
    const {
      day: dayClass,
      date: dateClass,
      container: dayContainer,
      dot: dayDot,
    } = this.props.customClass;

    return (
      <GridContainer
        spacing={0}
        alignItems="center"
        className={classNames(dayContainer)}
      >
        <GridItem>
          <H4 className={classNames(classes.day, dayClass)}>{day}</H4>
        </GridItem>
        <GridItem>
          <div className={classNames(classes.dot, dayDot)}>{THE_DOT}</div>
        </GridItem>
        <GridItem className={classes.grow}>
          <H4 className={classNames(classes.date, dateClass)}>{date}</H4>
          {displayInfo || ''}
        </GridItem>
      </GridContainer>
    );
  };

  renderDayAndInfo = (classes, day) => {
    const { displayInfo, customClass } = this.props;
    const { day: dayClass, container: dayContainer, dot: dayDot } = customClass;

    return (
      <GridContainer
        spacing={0}
        alignItems="center"
        className={classNames(dayContainer)}
      >
        <GridItem>
          <H4 className={classNames(classes.day, dayClass)}>{day}</H4>
        </GridItem>
        <GridItem>
          <div className={classNames(classes.dot, dayDot)}>{THE_DOT}</div>
        </GridItem>
        <GridItem>{displayInfo}</GridItem>
      </GridContainer>
    );
  };

  render = () => {
    const {
      classes,
      index,
      displayDate,
      displayInfo,
      startTime,
      dayDateFormat,
    } = this.props;

    const day = NODE_HELPERS.renderDayIndex(index);
    const date = NODE_HELPERS.renderDate(
      this.calculateDate(),
      displayDate,
      dayDateFormat,
    );

    switch (displayDate) {
      case dateDisplay.startDate:
      case dateDisplay.weekDay: {
        if (startTime) {
          return this.renderDayAndDate(classes, day, date);
        }
        return this.renderDayOnly(classes, day);
      }
      case dateDisplay.none: {
        if (displayInfo) {
          return this.renderDayAndInfo(classes, day);
        }
        return this.renderDayOnly(classes, day);
      }

      default:
        return this.renderDayOnly(classes, day);
    }
  };
}

DayDateDisplay.propTypes = {
  // hoc
  classes: PropTypes.object.isRequired,

  // parent
  id: PropTypes.number, // eslint-disable-line react/no-unused-prop-types
  index: PropTypes.number,
  customClass: PropTypes.object,
  dayDateFormat: PropTypes.string,
  displayInfo: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),

  // resaga value
  displayDate: PropTypes.string,
  startTime: PropTypes.string,
};

DayDateDisplay.defaultProps = {
  id: 0,
  index: 0,
  customClass: { day: '', date: '' },

  displayDate: null,
  startTime: null,
};

export default compose(
  withStyles(style, { name: 'DayDateDisplay' }),
  resaga(CONFIG_IDS),
  resaga(CONFIG),
)(DayDateDisplay);
