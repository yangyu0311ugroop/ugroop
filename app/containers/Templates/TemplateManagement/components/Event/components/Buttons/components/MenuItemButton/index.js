import GridContainer from 'components/GridContainer/index';
import GridItem from 'components/GridItem/index';
// import { withStyles } from 'components/material-ui';
import MenuItem from 'components/Popper/components/MenuItem';
import { EVENT_STORE_HELPERS } from 'datastore/eventStore/helpers';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { compose } from 'redux';
import resaga from 'resaga';
import { EVENT_CONSTANTS } from 'utils/constants/events';
import dotProp from 'dot-prop-immutable';
import VIcon from 'viewComponents/Icon';
import { CONFIG } from './config';
// import styles from './styles';

export class MenuItemButton extends PureComponent {
  newEvent = (e, props) => {
    const { id } = this.props;
    const {
      onOpen,
      type,
      subtype,
      iconOverride = {},
      typeOpen,
      subtypeOpen,
    } = props;

    this.props.resaga.setValue({
      selectedId: id,
      eventCreate: EVENT_STORE_HELPERS.setEventCreate(
        true,
        id,
        onOpen([type, subtype, iconOverride.type], typeOpen, subtypeOpen),
      ),
    });
  };

  openCreateEventDialog = (
    [formType, formSubtype, iconOverride = ''],
    typeOpen = false,
    subtypeOpen = false,
  ) => () => {
    this.props.resaga.setValue({
      formType,
      formSubtype,
      iconOverride,
    });

    return { typeOpen, subtypeOpen };
  };

  name = (event, iconOverride) => {
    const { name } = this.props;

    return name || iconOverride.name || event.name;
  };

  icon = (event, iconOverride) => iconOverride.icon || event.icon;

  color = event => event.color;

  renderAddEventMenuButton = (event, iconOverride) => {
    const name = this.name(event, iconOverride);
    const icon = this.icon(event, iconOverride);
    const color = this.color(event);
    const style = { color };

    // TODO: Probably remove if ever we decided to create our own icon for bicycle that is same with other transportation icons
    if (icon === 'lnr-bicycle') style.lineHeight = 0.5;

    return (
      <GridContainer alignItems="center" wrap="nowrap">
        <GridItem>
          <VIcon icon={icon} size="extraSmall" style={style} />
        </GridItem>
        <GridItem>{name}</GridItem>
      </GridContainer>
    );
  };

  render = () => {
    const {
      type,
      subtype,
      iconOverride,
      closeMenu,
      typeOpen,
      subtypeOpen,
      onClick,
    } = this.props;

    if (typeof onClick === 'function') {
      return (
        <MenuItem onClick={onClick} closeMenu={closeMenu}>
          {this.renderAddEventMenuButton(
            EVENT_CONSTANTS[type][subtype],
            iconOverride,
          )}
        </MenuItem>
      );
    }

    const t = dotProp.get(EVENT_CONSTANTS, `TYPES.${type}.type`);
    const st = dotProp.get(EVENT_CONSTANTS, `${type}.${subtype}.type`);

    return (
      <MenuItem
        onClick={this.newEvent}
        onOpen={this.openCreateEventDialog}
        type={t}
        subtype={st}
        iconOverride={iconOverride}
        closeMenu={closeMenu}
        typeOpen={typeOpen}
        subtypeOpen={subtypeOpen}
      >
        {this.renderAddEventMenuButton(
          EVENT_CONSTANTS[type][subtype],
          iconOverride,
        )}
      </MenuItem>
    );
  };
}

MenuItemButton.propTypes = {
  // hoc props
  resaga: PropTypes.object.isRequired,

  // parent props
  id: PropTypes.number, // event node id
  type: PropTypes.string,
  subtype: PropTypes.string,
  name: PropTypes.string,
  iconOverride: PropTypes.object,
  closeMenu: PropTypes.func,
  onClick: PropTypes.func,
  typeOpen: PropTypes.bool,
  subtypeOpen: PropTypes.bool,

  // resaga props
};

MenuItemButton.defaultProps = {
  iconOverride: {},
};

export default compose(
  // withStyles(styles, { name: 'MenuItemButton' }),
  resaga(CONFIG),
)(MenuItemButton);
