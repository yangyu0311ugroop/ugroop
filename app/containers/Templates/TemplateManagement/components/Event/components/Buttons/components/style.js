/**
 * Created by stephenkarpinskyj on 16/3/18.
 */

import {
  EVENT_CONSTANTS,
  UNKNOWN_EVENT_TYPE,
  UNKNOWN_EVENT_SUBTYPE,
  TRANSPORTATION_DEFAULT_COLOR,
  ACTIVITY_DEFAULT_COLOR,
  FOOD_DEFAULT_COLOR,
  ACCOMMODATION_COLORS,
} from 'utils/constants/events';
import { EVENT_HELPERS } from 'utils/helpers/events';
import {
  EVENT_ICON_SIZE,
  EVENT_ICON_SIZE_WITH_LABEL,
  EVENT_ICON_SIZE_XS,
} from '../../../constants';

export const makeEventIconStyles = () =>
  Object.values(EVENT_CONSTANTS.EVENTS).reduce(
    (acc, { type, getTypes }) => ({
      ...acc,
      ...Object.values(getTypes()).reduce(
        (acc2, { type: subtype, colors }) => ({
          ...acc2,
          ...makeEventIconStyleGroup(type, subtype, colors),
        }),
        {},
      ),
    }),
    {},
  );

export const makeUnknownEventIconStyles = () => {
  const { type } = UNKNOWN_EVENT_TYPE;
  const { type: subtype, colors } = UNKNOWN_EVENT_SUBTYPE;
  return makeEventIconStyleGroup(type, subtype, colors);
};

export const makeEventIconStyleGroup = (type, subtype, colors) => ({
  [EVENT_HELPERS.getEventSubtypeClassName(type, subtype)]: makeEventIconStyle(
    colors,
  ),
  [EVENT_HELPERS.getEventSubtypeHoverClassName(
    type,
    subtype,
  )]: makeEventIconHoverStyle(colors),
  [EVENT_HELPERS.getEventSubtypeClassBorderColorName(
    type,
    subtype,
  )]: maketriangleSolidColor(colors),
  [EVENT_HELPERS.getEventSubtypeClassSolidColorName(
    type,
    subtype,
  )]: maketriangleSolidColor(colors),
});

export const makeEventIconStyle = ({ icon }, invertColor) => ({
  // color: invertColor ? icon.background.default : invertColor,
  color: invertColor || icon.background.default,
  // border: invertColor
  //   ? `1px solid ${icon.background.default}`
  //   : '1px solid #fff',
  '&:hover': makeEventIconHoverStyle({ icon }, invertColor),
  // '&:hover i': makeEventIconHoverStyle({ icon }, invertColor),
});

export const makeEventIconHoverStyle = ({ icon }, invertColor) => ({
  // color: invertColor ? icon.background.hover : invertColor,
  backgroundColor: `${invertColor || icon.background.hover}`,
  color: 'white',
  boxShadow: 'unset',
  // border: `1px solid ${icon.background.hover}`,
});

export const maketriangleBorderColor = ({ icon }) => ({
  '&:before': {
    borderColor: `${icon.background.default} transparent`,
  },
});

export const maketriangleSolidColor = ({ icon }) => ({
  '&:after': {
    borderColor: `${icon.background.default} transparent`,
  },
});

const style = {
  // tooltip styles
  notClickable: {
    cursor: 'unset',
  },

  noTriangleStyle: {
    width: '100%',
    display: 'flex',
    alignItems: 'inherit',
    justifyContent: 'inherit',
    padding: '0 8px 0 8px',
  },

  checkmark: {
    fontWeight: '400',
    fontSize: '20px',
  },

  iconRoot: {
    color: 'white',
    position: 'absolute',
    zIndex: 2,
    top: -8,
    right: 0,
    backgroundColor: '#7aa1ea',
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    padding: '2px',
  },

  durationText: {
    fontWeight: '600',
    color: '#4C5673',
    fontStyle: 'italic',
  },
  duration: {
    color: '#4b5972',
  },

  description: {
    color: '#4b5972',
  },

  eventTitle: {
    margin: '0',
    fontWeight: '600',
  },

  hr: {
    margin: '8px 0',
    borderTopWidth: '1px',
    borderTopStyle: 'solid',
  },

  locationTitle: {
    color: '#cad1d9',
    marginBottom: '8px',
    fontSize: '12px',
    fontWeight: '600',
  },
  location: {
    color: '#7aa1ea',
    textDecoration: 'underline',
    margin: 0,
  },
  // end of tooltip styles
  button: {
    width: EVENT_ICON_SIZE,
    height: 32,
    borderRadius: '4px',
    color: 'white',
    zIndex: 1,
  },

  buttonHover: {},

  buttonLeftSideOffset: {
    marginLeft: -20,
  },

  triangle: {
    position: 'relative',
    '&:after': {
      content: '""',
      position: 'absolute',
      left:
        'calc(50% - 6px)' /* value = (:before left) + (:before border-left) - (:after border-left) */,
      bottom: '-8px',
      borderWidth: '10px 6px 0',
      borderStyle: 'solid',
      borderColor: '#fff transparent',
      display: 'block',
      width: 0,
    },
  },

  triangleBorder: {
    position: 'relative',
    '&:before': {
      content: '""',
      position: 'absolute',
      left:
        'calc(50% - 6px)' /* value = (:before left) + (:before border-left) - (:after border-left) */,
      bottom: '-8px',
      borderWidth: '10px 6px 0',
      borderStyle: 'solid',
      borderColor: '#f3961c transparent',
      display: 'block',
      width: 0,
    },
    '&:after': {
      content: '""',
      position: 'absolute',
      bottom: '-8px',
      left:
        'calc(50% - 6px)' /* value = (:before left) + (:before border-left) - (:after border-left) */,
      borderWidth: '10px 6px 0',
      borderStyle: 'solid',
      borderColor: '#fff transparent',
      display: 'block',
      width: 0,
    },
  },
  hidden: {
    display: 'none',
  },
  buttonSmall: {
    padding: '0px',
  },
  buttonSmallHover: {
    boxShadow: '0 6px 4px 0 #e3e9ef',
  },
  buttonWithLabel: {
    width: '100%',
  },
  eventLabel: {
    minWidth: 0,
    maxWidth: EVENT_ICON_SIZE_WITH_LABEL,
    marginLeft: 2,
    marginBottom: 0,
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    fontSize: 12,
    cursor: 'inherit',
    fontWeight: 'normal',
  },
  buttonXSmall: {
    minWidth: EVENT_ICON_SIZE_XS,
    width: EVENT_ICON_SIZE_XS,
    height: EVENT_ICON_SIZE_XS,
  },
  buttonAdd: {
    minWidth: 100,
    width: 100,
    height: EVENT_ICON_SIZE,
    background: '#a1c99c',

    '&:hover': {
      background: '#6ba380',
    },
  },
  buttonDragging: {
    boxShadow: '0 0 4px 4px #e3e9ef',
  },

  tooltip: {
    background: 'white',
    boxShadow: '0 3px 12px rgb(143 149 156)',
    padding: 0,
    fontWeight: 400,
  },
  popper: {
    opacity: 1,
    '& > div': {
      transformOrigin: 'left top',
    },
  },
  tooltipCard: {
    backgroundColor: 'white',
    fontWeight: 400,
    whiteSpace: 'normal',
    boxShadow: '0 1px 3px 0 rgba(37,32,31,.3)',
  },
  tooltipCardContent: {
    padding: 24,
  },
  headline: {
    '& h3': {
      textAlign: 'center',
    },
  },
  attachmentContent: {
    backgroundColor: '#f8f8fa',
    color: '#4C5673',
    marginTop: 0,
    fontWeight: 400,
    zDepth: 2,
    whiteSpace: 'normal',
  },

  Add: {
    color: '#8abd84',
    backgroundColor: '#fff',
    margin: '0px 1px 0px',
    padding: '0',
    '&:hover': {
      color: '#8abd84',
    },
  },

  ...makeEventIconStyles(),
  ...makeUnknownEventIconStyles(),

  circle: {
    borderRadius: '50%',
  },

  checkOut: {
    ...makeEventIconStyle(
      EVENT_HELPERS.getEventSubtypeConstants(
        EVENT_CONSTANTS.EVENTS.ACCOMMODATION.type,
        EVENT_CONSTANTS.ACCOMMODATIONS.HOTEL.type,
      ).colors,
    ),
  },

  landing: {
    ...makeEventIconStyle(
      EVENT_HELPERS.getEventSubtypeConstants(
        EVENT_CONSTANTS.EVENTS.TRANSPORTATION.type,
        EVENT_CONSTANTS.TRANSPORTATIONS.FLIGHT.type,
      ).colors,
    ),
  },

  coachDropoff: {
    ...makeEventIconStyle(
      EVENT_HELPERS.getEventSubtypeConstants(
        EVENT_CONSTANTS.EVENTS.TRANSPORTATION.type,
        EVENT_CONSTANTS.TRANSPORTATIONS.COACH.type,
      ).colors,
    ),
  },

  busDropoff: {
    ...makeEventIconStyle(
      EVENT_HELPERS.getEventSubtypeConstants(
        EVENT_CONSTANTS.EVENTS.TRANSPORTATION.type,
        EVENT_CONSTANTS.TRANSPORTATIONS.BUS.type,
      ).colors,
    ),
  },
  vehiclehireDropoff: {
    ...makeEventIconStyle(
      EVENT_HELPERS.getEventSubtypeConstants(
        EVENT_CONSTANTS.EVENTS.TRANSPORTATION.type,
        EVENT_CONSTANTS.TRANSPORTATIONS.VEHICLE_HIRE.type,
      ).colors,
    ),
  },
  buttonFade: {
    opacity: 0.5,
  },
  confirmed: {
    background: 'white',
  },
  'Transportation.confirmedHover': {
    background: TRANSPORTATION_DEFAULT_COLOR.icon.background.default,
    color: 'white',
  },
  'Activity.confirmedHover': {
    background: ACTIVITY_DEFAULT_COLOR.icon.background.default,
    color: 'white',
  },
  'Accommodation.confirmedHover': {
    background: ACCOMMODATION_COLORS.icon.background.default,
    color: 'white',
  },
  'Food.confirmedHover': {
    background: FOOD_DEFAULT_COLOR.icon.background.default,
    color: 'white',
  },

  showButton: {
    border: '1px solid gainsboro',
    background: 'white',
  },

  addBtn: {
    '&:hover': {
      borderStyle: 'solid',
      borderWidth: '1px',
      borderColor: '#a1c99c',
      paddingBottom: '34px',
    },
  },
};

export default style;
