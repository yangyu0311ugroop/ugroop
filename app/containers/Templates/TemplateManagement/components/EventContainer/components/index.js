/**
 * Created by stephenkarpinskyj on 20/8/18.
 */

export EventCreate from './Event/Create';
export EventView from './Event/View';
export EventEdit from './Event/Edit';

export FlightBookingCreate from './FlightBooking/Create';
export FlightBookingView from './FlightBooking/View';
export FlightBookingEdit from './FlightBooking/Edit';
