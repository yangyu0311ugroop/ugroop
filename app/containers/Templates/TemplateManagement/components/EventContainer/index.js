/**
 * Created by stephenkarpinskyj on 24/3/18.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import resaga from 'resaga';

import { EVENT_STORE_HELPERS } from 'datastore/eventStore/helpers';
import {
  EventCreate,
  EventView,
  EventEdit,
  FlightBookingCreate,
  FlightBookingView,
  FlightBookingEdit,
} from './components';
import { CONFIG } from './config';

export class EventContainer extends React.PureComponent {
  handleCloseEventCreate = () => {
    const { createDayId } = this.props;
    this.props.resaga.setValue({
      eventCreate: EVENT_STORE_HELPERS.setEventCreate(false, createDayId),
    });
  };

  handleCloseEventView = () => {
    const { viewId } = this.props;
    this.props.resaga.setValue({
      eventView: EVENT_STORE_HELPERS.setEventView(false, viewId),
    });
  };

  handleCloseEventEdit = () => {
    const { editId } = this.props;
    this.props.resaga.setValue({
      eventEdit: EVENT_STORE_HELPERS.setEventEdit(false, editId),
    });
  };

  handleCloseFlightBookingCreate = dataId => {
    const { flightBookingCreateHandler } = this.props;
    this.props.resaga.setValue({
      flightBookingCreate: EVENT_STORE_HELPERS.setFlightBookingCreate(false),
    });
    flightBookingCreateHandler(dataId ? dataId.toString() : dataId);
  };

  handleCloseFlightBookingView = () => {
    const { flightBookingViewId } = this.props;
    this.props.resaga.setValue({
      flightBookingView: EVENT_STORE_HELPERS.setFlightBookingView(
        false,
        flightBookingViewId,
      ),
    });
  };

  handleCloseFlightBookingEdit = () => {
    const { flightBookingEditId } = this.props;
    this.props.resaga.setValue({
      flightBookingEdit: EVENT_STORE_HELPERS.setFlightBookingEdit(
        false,
        flightBookingEditId,
      ),
    });
  };

  renderDialog = (Component, props) => <Component {...props} />;

  renderFlightBooking = () => {
    const {
      flightBookingCreateOpen,
      flightBookingCreateTime,
      flightBookingViewOpen,
      flightBookingViewId,
      flightBookingViewTime,
      flightBookingEditOpen,
      flightBookingEditId,
      flightBookingEditTime,
    } = this.props;
    return (
      <React.Fragment>
        {this.renderDialog(FlightBookingCreate, {
          open: flightBookingCreateOpen,
          time: flightBookingCreateTime,
          onClose: this.handleCloseFlightBookingCreate,
        })}
        {this.renderDialog(FlightBookingView, {
          open: flightBookingViewOpen,
          dataId: flightBookingViewId,
          time: flightBookingViewTime,
          onClose: this.handleCloseFlightBookingView,
        })}
        {this.renderDialog(FlightBookingEdit, {
          open: flightBookingEditOpen,
          dataId: flightBookingEditId,
          time: flightBookingEditTime,
          onClose: this.handleCloseFlightBookingEdit,
        })}
      </React.Fragment>
    );
  };

  renderEvent = () => {
    const {
      createOpen,
      createDayId,
      createTime,
      createOpenHandler,
      viewOpen,
      viewId,
      viewTime,
      editOpen,
      editId,
      editTime,
      ...props
    } = this.props;
    return (
      <React.Fragment>
        {this.renderDialog(EventCreate, {
          open: createOpen,
          dayId: createDayId,
          time: createTime,
          onOpen: createOpenHandler,
          onClose: this.handleCloseEventCreate,
          ...props,
        })}
        {this.renderDialog(EventView, {
          open: viewOpen,
          id: viewId,
          time: viewTime,
          onClose: this.handleCloseEventView,
        })}
        {this.renderDialog(EventEdit, {
          open: editOpen,
          id: editId,
          time: editTime,
          onClose: this.handleCloseEventEdit,
        })}
      </React.Fragment>
    );
  };

  render = () => (
    <React.Fragment>
      {this.renderEvent()}
      {this.renderFlightBooking()}
    </React.Fragment>
  );
}

EventContainer.propTypes = {
  // hoc
  resaga: PropTypes.object.isRequired,

  // resaga value
  createOpen: PropTypes.bool,
  createDayId: PropTypes.number,
  createTime: PropTypes.number,
  createOpenHandler: PropTypes.func,
  viewOpen: PropTypes.bool,
  viewId: PropTypes.number,
  viewTime: PropTypes.number,
  editOpen: PropTypes.bool,
  editId: PropTypes.number,
  editTime: PropTypes.number,
  flightBookingCreateOpen: PropTypes.bool,
  flightBookingCreateHandler: PropTypes.func,
  flightBookingCreateTime: PropTypes.number,
  flightBookingViewOpen: PropTypes.bool,
  flightBookingViewId: PropTypes.number,
  flightBookingViewTime: PropTypes.number,
  flightBookingEditOpen: PropTypes.bool,
  flightBookingEditId: PropTypes.number,
  flightBookingEditTime: PropTypes.number,
};

EventContainer.defaultProps = {
  createOpen: false,
  createDayId: null,
  viewOpen: false,
  viewId: null,
  editOpen: false,
  editId: null,
  flightBookingCreateOpen: false,
  flightBookingCreateHandler: () => {},
  flightBookingViewOpen: false,
  flightBookingViewId: null,
  flightBookingEditOpen: false,
  flightBookingEditId: null,
};

export default compose(resaga(CONFIG))(EventContainer);
