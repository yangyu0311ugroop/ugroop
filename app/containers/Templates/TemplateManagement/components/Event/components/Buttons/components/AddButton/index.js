/**
 * Created by stephenkarpinskyj on 3/4/18.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { withStyles } from '@material-ui/core/styles';
import classnames from 'classnames';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { VARIANTS } from 'variantsConstants';
import Button from 'viewComponents/Button';

import style from '../style';

// eslint-disable-next-line react/prefer-stateless-function
export class UGEventAddButton extends PureComponent {
  getOtherProps = () => ({
    className: LOGIC_HELPERS.ifElse(
      this.props.location === 'CARD_VIEW',
      '',
      classnames(
        this.props.classes.button,
        this.props.classes.Add,
        this.props.classes.addBtn,
        this.props.className,
      ),
    ),
    dense: LOGIC_HELPERS.ifElse(
      this.props.location === 'CARD_VIEW',
      true,
      false,
    ),
  });

  handleButtonClick = e => {
    e.preventDefault();
    this.props.onClick();
  };

  render = () => (
    <Button
      square
      size="small"
      title="Add Event"
      variant={VARIANTS.BORDERLESS}
      dense
      color="primary"
      icon="plus"
      iconButton
      disableRipple
      onClick={this.handleButtonClick}
      {...this.getOtherProps()}
    />
  );
}

UGEventAddButton.propTypes = {
  // hoc
  classes: PropTypes.object.isRequired,

  // parent
  onClick: PropTypes.func.isRequired,
  location: PropTypes.string,
  className: PropTypes.string,
};

UGEventAddButton.defaultProps = {
  location: '',
};

export default compose(withStyles(style, { name: 'UGEventTooltipIconButton' }))(
  UGEventAddButton,
);
