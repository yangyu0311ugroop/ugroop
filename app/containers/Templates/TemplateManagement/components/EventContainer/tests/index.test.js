/**
 * Created by stephenkarpinskyj on 24/10/18.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { EventContainer } from '..';
import * as components from '../components';

describe('<EventContainer />', () => {
  let wrapper;
  let instance;
  let doResagaSnapshot;

  const makeProps = () => ({
    resaga: {
      setValue: jest.fn(
        obj => doResagaSnapshot && expect({ setValue: obj }).toMatchSnapshot(),
      ),
    },
    createDayId: 1,
    viewId: 2,
    editId: 3,
    flightBookingViewId: 4,
    flightBookingEditId: 5,
  });

  Date.now = jest.fn(() => 'now');

  beforeEach(() => {
    wrapper = shallow(<EventContainer {...makeProps()} />);
    instance = wrapper.instance();
    doResagaSnapshot = false;
    jest.clearAllMocks();
  });

  it('exists', () => {
    expect(EventContainer).toBeDefined();
  });

  describe('#handleCloseEventCreate()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseEventCreate();
    });
  });

  describe('#handleCloseEventView()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseEventView();
    });
  });

  describe('#handleCloseEventEdit()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseEventEdit();
    });
  });

  describe('#handleCloseFlightBookingCreate()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseFlightBookingCreate(0);
    });

    it('flightBookingCreateHandler is called with id', () => {
      const flightBookingCreateHandler = jest.fn();
      wrapper.setProps({ flightBookingCreateHandler });
      instance.handleCloseFlightBookingCreate(1);
      expect(flightBookingCreateHandler).toBeCalledWith('1');
    });
  });

  describe('#handleCloseFlightBookingView()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseFlightBookingView();
    });
  });

  describe('#handleCloseFlightBookingEdit()', () => {
    it('resaga.setValue still matches snapshot', () => {
      doResagaSnapshot = true;
      instance.handleCloseFlightBookingEdit();
    });
  });

  describe('#render()', () => {
    it('not explodes', () => {
      expect(wrapper).toHaveLength(1);
    });

    it('still matches snapshot', () => {
      instance.handleCloseEventCreate = 'handleCloseEventCreate';
      instance.handleCloseEventView = 'handleCloseEventView';
      instance.handleCloseEventEdit = 'handleCloseEventEdit';
      instance.handleCloseFlightBookingCreate =
        'handleCloseFlightBookingCreate';
      instance.handleCloseFlightBookingEdit = 'handleCloseFlightBookingEdit';

      components.EventCreate.displayName = 'EventCreate';
      components.EventView.displayName = 'EventView';
      components.EventEdit.displayName = 'EventEdit';
      components.FlightBookingCreate.displayName = 'FlightBookingCreate';
      components.FlightBookingEdit.displayName = 'FlightBookingEdit';

      expect(instance.render()).toMatchSnapshot();
    });
  });

  describe('#defaultProps', () => {
    it('#flightBookingCreateHandler()', () => {
      expect(() => {
        EventContainer.defaultProps.flightBookingCreateHandler();
      }).not.toThrow();
    });
  });
});
