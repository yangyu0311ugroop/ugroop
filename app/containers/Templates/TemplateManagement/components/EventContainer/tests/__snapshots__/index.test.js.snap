// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventContainer /> #handleCloseEventCreate() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "eventCreate": Object {
      "createDayId": 1,
      "createOpen": false,
      "createOpenHandler": undefined,
      "createTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #handleCloseEventEdit() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "eventEdit": Object {
      "editId": 3,
      "editOpen": false,
      "editTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #handleCloseEventView() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "eventView": Object {
      "viewId": 2,
      "viewOpen": false,
      "viewTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #handleCloseFlightBookingCreate() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "flightBookingCreate": Object {
      "flightBookingCreateHandler": undefined,
      "flightBookingCreateOpen": false,
      "flightBookingCreateTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #handleCloseFlightBookingEdit() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "flightBookingEdit": Object {
      "flightBookingEditId": 5,
      "flightBookingEditOpen": false,
      "flightBookingEditTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #handleCloseFlightBookingView() resaga.setValue still matches snapshot 1`] = `
Object {
  "setValue": Object {
    "flightBookingView": Object {
      "flightBookingViewId": 4,
      "flightBookingViewOpen": false,
      "flightBookingViewTime": "now",
    },
  },
}
`;

exports[`<EventContainer /> #render() still matches snapshot 1`] = `
<React.Fragment>
  <React.Fragment>
    <EventCreate
      dayId={1}
      flightBookingCreateHandler={[Function]}
      flightBookingCreateOpen={false}
      flightBookingEditId={5}
      flightBookingEditOpen={false}
      flightBookingViewId={4}
      flightBookingViewOpen={false}
      onClose="handleCloseEventCreate"
      open={false}
      resaga={
        Object {
          "setValue": [MockFunction],
        }
      }
    />
    <EventView
      id={2}
      onClose="handleCloseEventView"
      open={false}
    />
    <EventEdit
      id={3}
      onClose="handleCloseEventEdit"
      open={false}
    />
  </React.Fragment>
  <React.Fragment>
    <FlightBookingCreate
      onClose="handleCloseFlightBookingCreate"
      open={false}
    />
    <WithStyles(Key)
      dataId={4}
      onClose={[Function]}
      open={false}
    />
    <FlightBookingEdit
      dataId={5}
      onClose="handleCloseFlightBookingEdit"
      open={false}
    />
  </React.Fragment>
</React.Fragment>
`;
